boilerplate/update

```diff
@@ -14,41 +14,101 @@ if [ "$BOILERPLATE_SET_X" ]; then
     set -x
 fi
 
-DIR="$(realpath $(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd ))"
-
-# Were we re-exec'd?
-if [ "$1" ]; then
-  OUT=$1
-  # Let's make sure this was really a re-exec and not some other
-  # weirdness.
-  if ! [[ -d "$OUT" ]] || ! diff -q "$0" "$OUT/boilerplate/update"; then
-    echo "Something went wrong! I was invoked with '$OUT'!"
+# The directory in which this script lives is the CONVENTION_ROOT. Export
+# this for individual `update` scripts.
+export CONVENTION_ROOT="$(realpath $(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd ))"
+CONFIG_FILE="${CONVENTION_ROOT}/update.cfg"
+
+# scrubbed_conventions
+# Parses the $CONFIG_FILE and outputs a space-delimited list of conventions.
+scrubbed_conventions() {
+  local conventions=
+
+  while read convention junk; do
+    # Skip comment lines (which can have leading whitespace)
+    if [[ "$convention" == '#'* ]]; then
+      continue
+    fi
+
+    # Skip blank or whitespace-only lines
+    if [[ "$convention" == "" ]]; then
+      continue
+    fi
+
+    # Lines like
+    #    valid/path  other_junk
+    # are not acceptable, unless `other_junk` is a comment
+    if [[ "$junk" != "" ]] && [[ "$junk" != '#'* ]]; then
+      echo "Invalid config! Only one convention is allowed per line. Found '$junk'"
+      exit 1
+    fi
+
+    # This one is legit
+    echo $convention
+
+  done < "$CONFIG_FILE"
+}
+
+# unacceptable_deltas [GREP_FLAGS]
```

Comment:
Could be good to describe the aim of the function

Reply:
Will do.

Reply:
Done.

While doing so, I of course found myself having to explain each of the pieces, and I couldn't figure out the real justification for go.mod, go.sum, and the convention ignore list. So I'm removing those.

Reply:
Update: It was a failure in the generator test, which stemmed from the fact that that test wasn't using the "proper" workflow for bootstrapping. (This happened because the generator test and this PR were in flight at the same time.) I updated the test and ripped these out. #62.